### ISSUE TITLE
[Switch]$NoWait = $false, must be   [Boolean]$NoWait = $false, Breaks function * Show-InstallationPrompt version 3.10

### ISSUE THREAD

**Issue by djalexnl** [PSAppDeployToolkit/PSAppDeployToolkit#929]
**Describe the bug**
[Switch]$NoWait = $false, must be   [Boolean]$NoWait = $false, Breaks function * Show-InstallationPrompt

**Toolkit Version:**
Version of our toolkit you are using.  [3.10.0](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/commit/389213775a443909d60d457a9cef29f71a865a38)

Change this back 2 [Boolean]$NoWait = $false, (as it has always been) and all is OK ;-)

**Comment by That-Annoying-Guy** [comment]
RE: As a sidenote, there is a confusing mix of switch vs boolean use throughout PSADT, and it's something I'd like us to look at in 4.0!

Correct me if I'm wrong but I could have sworn not being able to splat a $false to a switch. Splats only work with Boolean.

**Comment by mjr4077au** [comment]
Splats work fine with switches, so there's definitely no reason to use booleans for parameter arguments. Maybe an edge case is when you're doing something via batch scripts, but certainly edge case.

**Comment by DanGough** [comment]
I agree that splatting switches is fine, the only issue with splatting I've seen is not being able to send null values to booleans that have ValidateNotNullOrEmpty set. It also doesn't make any sense to me to have validation on any switch parameters.

**Comment by mjr4077au** [comment]
Makes no sense to me either, that's just a bad setup. Parameters should only be decorated with validation attributes when they accept actual input.

**Comment by sintaxasn** [comment]
Regardless, I'm afraid we're stuck with them until after v4. But we can identify and scope these sort of confusing param names and incorrect types in advance. And from there we can figure out how to fix and piece together a mechanism for upgrading existing scripts with the required changes - then take one big swing at them after 4.0.

**Comment by djalexnl** [comment]
> That line was changed 15 months ago - how is it breaking for you exactly?
> 
> If you are finding that it breaks when specifying `-NoWait $true` for example, then just drop the $true; and if you want it to be false, do not supply the switch. For switches you can also use notation like `-NoWait:$SwitchValue` if you ever need to pass an existing boolean to the function call.
> 
> A couple of examples that work:
> 
> ```postscript
> Show-InstallationPrompt -Title 'Test' -Message 'Is this working?' -ButtonRightText 'Yes' -ButtonLeftText 'Yes' -NoWait
> ```
> 
> ```postscript
> $NoWaitValue = $true
> Show-InstallationPrompt -Title 'Test' -Message 'Is this working?' -ButtonRightText 'Yes' -ButtonLeftText 'Yes' -NoWait:$NoWaitValue
> ```
> 
> As a sidenote, there is a confusing mix of switch vs boolean use throughout PSADT, and it's something I'd like us to look at in 4.0!

I switched to -NoWait and it works perfectly, as expected. It's odd because I've been using this toolkit for 3 months and thought I had the latest version back then, so I was somewhat surprised to see this change. Nevertheless, I've updated my fork, so thanks for pointing this out.
